google/cloud/spanner/client.h

```diff
@@ -0,0 +1,193 @@
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_SPANNER_CLIENT_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_SPANNER_CLIENT_H_
+
+#include "google/cloud/spanner/key.h"
+#include "google/cloud/spanner/mutation.h"
+#include "google/cloud/spanner/partition.h"
+#include "google/cloud/spanner/result_set.h"
+#include "google/cloud/spanner/result_stats.h"
+#include "google/cloud/spanner/row.h"
+#include "google/cloud/spanner/sql_statement.h"
+#include "google/cloud/spanner/transaction.h"
+#include "google/cloud/spanner/value.h"
+#include "google/cloud/spanner/version.h"
+#include "google/cloud/status.h"
+#include "google/cloud/status_or.h"
+#include <chrono>
+#include <string>
+#include <vector>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+
+class Client {
+ public:
+  // move-only
+  Client(Client&&) = default;
+  Client& operator=(Client&&) = default;
+  Client(Client const&) = delete;
+  Client& operator=(Client const&) = delete;
+
+  //
+  // Read()
+  //
+
+  // Reads the columns for the given keys from the specified table. Returns a
```

Comment:
Need to add doxygen blocks to each one of these functions. That may be too much for this PR, but please add a bug to say so.